spec/models/representation_order_spec.rb

```diff
@@ -52,13 +53,35 @@
         expect(representation_order.errors[:maat_reference]).to eq( [ 'invalid'])
       end
 
-      it 'should not error if 7-10 numeric digits' do
-        representation_order.maat_reference = '2078352232'
+      it 'should not error if 7 numeric digits' do
+        representation_order.maat_reference = '5078332'
         expect(representation_order).to be_valid
       end
+
+      context 'our test MAAT reference' do
+        before { representation_order.maat_reference = '2320006' }
+
+        context 'when on the live environment' do
+          before do
+            allow(ENV).to receive(:[]).with('MAAT_REGEXP').and_return /^[4-9][0-9]{6}$/
+            allow(Settings).to receive(:maat_regexp).and_return ENV['MAAT_REGEXP']
```

Comment:
is this second mock required? i know rspec-mocks get funny, but would hope the ENV stub would be sufficient

Reply:
It was because the Settings gem doesn't actually read the env vars in test, so I had to mock the env var first and then simulate `Settings` reading it.
I have refactored, and renamed the test, as the actual function changed over time.  Does the new version seem better to you?

Reply:
yep 👍